### PR TITLE
Add e2e to validate stats for Windows hostprocess containers

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add e2e to validate stats for Windows hostprocess containers

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig windows

### CODE REVIEW

**Comment by jsturtevant** [comment]
This will require a fix in hcsshim to pass: https://github.com/microsoft/hcsshim/pull/1465

**Comment by marosset** [comment]
> This will require a fix in hcsshim to pass: [microsoft/hcsshim#1465](https://github.com/microsoft/hcsshim/pull/1465)

Correct, we've found that containrd v1.6.5 and v1.6.6 have issues reporting stats for host-process containers.
I manually validated these tests by setting up a cluster with containerd v1.6.4 on the Windows nodes.
We can keep this PR open until this issue gets patched.

**Comment by marosset** [comment]
/assign @dcantah 
/hold 
for to confirm test ran and passed in `pull-kubernetes-e2e-capz-windows-containerd`

**Comment by marosset** [comment]
The new test passed in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111332/pull-kubernetes-e2e-capz-windows-containerd/1570474265726160896

/hold cancel
/assign @dcantah @jsturtevant

**Comment by marosset** [comment]
@jsturtevant - PTAL

**Comment by jsturtevant** [comment]
the new test passed:

```
    {"msg":"PASSED [sig-windows] [Feature:WindowsHostProcessContainers] [MinimumKubeletVersion:1.22] HostProcess containers container stats validation","completed":7,"skipped":190,"failed":0}
```

/lgtm

**Comment by pacoxu** [comment]
/hold
```
test/e2e/windows/host_process.go:637:5: f.PodClient undefined (type *framework.Framework has no field or method PodClient)
```
You need to fix the ci and feel free to cancel the hold.

File: test/e2e/windows/host_process.go [hunk 1]
```diff
@@ -600,6 +601,84 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
+						Name:    "host-process-stats",
+						Command: []string{"powershell.exe", "-Command", "Write-Host 'Hello'; sleep -Seconds 600"},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+				NodeName:      targetNode.Name,
+			},
+		}
+
+		f.PodClient().Create(pod)
```
Comment (pacoxu) [inline test/e2e/windows/host_process.go:637]:
```suggestion
		e2epod.NewPodClient(f).Create(pod)
```

Comment (marosset) [reply test/e2e/windows/host_process.go:637]: I'll rebase and clean this up.
Thanks!

**Comment by marosset** [comment]
/test pull-kubernetes-e2e-capz-windows-containerd
infra flake

**Pull request merged**